src/com/amazon/corretto/crypto/provider/MlDsaGen.java

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+import java.security.KeyPair;
+import java.security.KeyPairGeneratorSpi;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+
+class MlDsaGen extends KeyPairGeneratorSpi {
+  /** Generates a new ML-DSA key and returns a pointer to it. */
+  private static native long generateEvpMlDsaKey(int nid);
+
+  private final AmazonCorrettoCryptoProvider provider_;
+  private int type_ = -1;
```

Comment:
The nomenclature around this value is all over the place. I see the names `type`, `nid`, and `level` to refer to this integer. And I can't really work out why we're using the values 2, 3, and 5 for it.

Reply:
fair. will standardize this on `level`, which refers to "claimed NIST level" [here](https://openquantumsafe.org/liboqs/algorithms/sig/ml-dsa.html).